bin/PerlDDS/Run_Test.pm

```diff
@@ -798,14 +805,24 @@ sub stop_process {
     }
   }
 
-  my $kill_status =
-    PerlDDS::wait_kill($self->{processes}->{process}->{$name}->{process},
-                       $timed_wait,
-                       $name,
-                       $self->{test_verbose});
+  my @ret_array;
+  my $kill_status;
+  if (wantarray()) {
+    @ret_array = PerlDDS::wait_kill($self->{processes}->{process}->{$name}->{process},
+                                    $timed_wait,
+                                    $name,
+                                    $self->{test_verbose});
+    $kill_status = $ret_array[0];
+  } else {
+    $kill_status = PerlDDS::wait_kill($self->{processes}->{process}->{$name}->{process},
+                                      $timed_wait,
+                                      $name,
+                                      $self->{test_verbose});
+  }
```

Comment:
I'm not sure about this now that I see it. It seems like it would be cleaner to go with the hash ref method I showed in the other comment. That way we wouldn't have to double everything up.